homeassistant/components/stream/hls.py

```diff
@@ -95,11 +99,12 @@ def render_playlist(track):
             # at the beginning or we risk a behind live window exception in exoplayer.
             # EXT-X-START is not supposed to be within 3 target durations of the end,
             # but this seems ok
-            f"#EXT-X-START:TIME-OFFSET=-{EXT_X_START * track.target_duration:.3f},PRECISE=YES",
+            f"#EXT-X-START:TIME-OFFSET=-{EXT_X_START * track.target_duration:.3f}",
         ]
 
         last_stream_id = first_segment.stream_id
-        for segment in segments:
+        # Add playlist sections for completed segments
+        for segment in segments if segments[-1].complete else segments[:-1]:
```

Comment:
It's actually a little different.
First there is a truncation at line 79 to remove the "extra" segment at the head, shrinking the deque to a list with a (max) size of 4.
The check/truncation on L83-L84 is to enforce that we have at most 3 complete segments by removing the first segment from the deque if we happen to have 4. However, those 3 segments could be at locations [-3,-2,-1] or at locations [-4,-3,-2] depending on if -1 is complete or not.
This check on this line is to avoid writing a playlist section for the index -1 if we have the [-4,-3,-2] case.
Sorry if the code or explanation is confusing. Maybe we can still clean this up.

Reply:
Ah i was just reading in a hurry -- no problem, something to consider in the future.

Reply:
Changed it from truncating the iterable to using an explicit if statement with a comment. Should be more clear.